paderbox/array/kernel.py

```diff
@@ -0,0 +1,212 @@
+import typing
+import numpy as np
+import paderbox as pb
+
+
+__all__ = [
+    'np_kernel',
+    'ai_dilate',
+    'ai_erode',
+    'max_kernel',
+    'min_kernel',
+    'mean_kernel',
+    'median_kernel',
+]
+
+
+def _plot(a):
+    """
+    >>> _plot([0, 1, 2, 3, 4, 5, 6, 7, 8])
+    ' ▁▂▃▄▅▆▇█'
+    """
+    a = np.array(a)
+    values = np.array(list(' ▁▂▃▄▅▆▇█'))
+    m = np.amax(a)
+    if m == 0:
+        m = 1
+    a = np.round(a * ((len(values)-1) / m)).astype(int)
+    print(repr(''.join(values[a])))
+
+
+def np_kernel1d(
+        x,
+        kernel_size,  # an odd number
+        *,
+        kernel: callable,  # e.g. np.mean, np.amax, np.amin
+        padding_mode: 'typing.Literal["edge"]' = 'edge',
+        pad_position: 'typing.Literal["pre", "post", None]' = 'pre',
+):
+    """
+    Apply a kernel to the last axis of x.
+
+    Args:
+        x: np.array
+        kernel_size:
+        kernel: a function that should be applied to the kernel window.
+            Has to accept an axis argument, like np.mean, np.amax, np.median.
+        padding_mode:
+            The mode for np.pad, see np.pad for the options.
+        pad_position:
```

Comment:
I was confused by the parameter name. This sounds like: `pre`: pad beginning, `post`: pad end. I don't have a strong opinion, but maybe just `pad`?

Reply:
In the other paderbox functions, `pad` always is a boolean parameter that defines whether there ist padding. So, I would not use that name here.

Reply:
Should it then be something like `edge in ['pre_pad', 'post_pad', 'cut']`?

I think `pad` is too general. IMHO the `position` triggers the user to check the documentation.
Or is there a better name for the execution order?

At the moment, I have a favor for keeping the name.

Reply:
I didn't find a better alternative for position, unfortunately. I don't like `edge` either. Then keep the name